modules/core/src/main/scala/GuardrailAnalysis.scala

```diff
@@ -0,0 +1,16 @@
+package com.twilio.guardrail
+package sbt
+
+import _root_.sbt._
+import _root_.sbt.util.CacheImplicits._
+import sjsonnew.{ :*:, LList, LNil}
+
+case class GuardrailAnalysis(products: List[java.io.File]) {
+  def ++(that: GuardrailAnalysis): GuardrailAnalysis =
+    GuardrailAnalysis(products ++ that.products)
+}
+object GuardrailAnalysis {
+  implicit val analysisIso = LList.iso(
+    { a: GuardrailAnalysis => ("products", a.products) :*: LNil },
+    { in: List[java.io.File] :*: LNil => GuardrailAnalysis(in._1) })
```

Comment:
Does `:*:` have an `unapply`?

```
@ case class :*:[A](a: A, b: A)
defined class :*:

@ implicit class syntax[A](a: A) { def :*:(b: A): :*:[A] = new :*:[A](a, b) }
defined class syntax

@ 5 :*: 3
res2: :*:[Int] = 3 :*: 5

@ val (a :*: b) = 5 :*: 3
a: Int = 3
b: Int = 5
```

Maybe this could remove the need for the `_1`?

Reply:
(actually, I realize this would end up needing to be `case b :*: a => ...` due to `:*:` being right-associative)

Reply:
(maybe that bit of confusion alone would be sufficient to keep the `_1`, what do you think?)

Reply:
attempted to improve the syntax, what do you think about this?